presto-native-execution/presto_cpp/main/common/Configs.cpp

```diff
@@ -718,14 +732,34 @@ BaseVeloxQueryConfig* BaseVeloxQueryConfig::instance() {
   return instance.get();
 }
 
-void BaseVeloxQueryConfig::initialize(const std::string& filePath) {
-  ConfigBase::initialize(filePath);
-  update(*SystemConfig::instance());
-}
+void BaseVeloxQueryConfig::updateLoadedValues(
+    std::unordered_map<std::string, std::string>& values) const {
+  /// Update velox config with values from presto system config.
+  auto systemConfig = SystemConfig::instance();
+
+  using namespace velox::core;
+  const std::unordered_map<std::string, std::string> updatedValues{
+      {QueryConfig::kPrestoArrayAggIgnoreNulls,
+       bool2String(systemConfig->useLegacyArrayAgg())},
+      {QueryConfig::kMaxArbitraryBufferSize,
+       systemConfig->capacityPropertyAsBytesString(
+           SystemConfig::kSinkMaxBufferSize)},
+      {QueryConfig::kMaxPartitionedOutputBufferSize,
+       systemConfig->capacityPropertyAsBytesString(
+           SystemConfig::kDriverMaxPagePartitioningBufferSize)},
+  };
 
-void BaseVeloxQueryConfig::update(const SystemConfig& systemConfig) {
-  registeredProps_[velox::core::QueryConfig::kPrestoArrayAggIgnoreNulls] =
-      bool2String(systemConfig.useLegacyArrayAgg());
+  std::stringstream updated;
+  for (const auto& pair : updatedValues) {
+    updated << "    " << pair.first << "=" << pair.second << "\n";
+    values[pair.first] = pair.second;
+  }
+  auto str = updated.str();
+  if (!str.empty()) {
+    PRESTO_STARTUP_LOG(INFO)
+        << "  Updated in '" << filePath_ << "' from SystemProperties:\n"
```

Comment:
This function is not specifically used for `systemConfig` right? it's also used for `nodeConfig` and `baseVeloxQueryConfig`

Reply:
`BaseVeloxQueryConfig::updateLoadedValues()` is added in this PR and is only utilized for BaseVeloxQueryConfig at the moment. To move some presto config props into the BaseVeloxQueryConfig props.

BaseVeloxQueryConfig is used as the base for all instances of QueryConfig.